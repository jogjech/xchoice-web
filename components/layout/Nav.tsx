import React, { FunctionComponent } from "react";
import { Layout, Menu } from "antd";
import Link from "next/link";
import styles from "./Nav.module.css";

const { Header } = Layout;

interface Props {}

const Nav: FunctionComponent<Props> = () => {
  return (
    <Header>
      <Link href="/">
        <div className={styles.logo}>
          <a>X Choice</a>
        </div>
      </Link>
      <Menu theme="dark" mode="horizontal">
        <Menu.Item>
          <Link href="/survey/create">
            <a>Create Survey</a>
          </Link>
        </Menu.Item>
        <Menu.Item>
          <Link href="/dashboard">
            <a>My Surveys</a>
          </Link>
        </Menu.Item>
      </Menu>
    </Header>
  );
};

export default Nav;
