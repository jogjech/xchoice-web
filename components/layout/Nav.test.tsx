import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Nav from "./Nav";

describe("Nav bar", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav></Nav>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains menu", () => {
    const menu = wrapper.find("Menu");
    expect(menu.exists()).toBeTruthy();
    const menuItems = menu.children();
    const links = menuItems.map((menuItem) => menuItem.children());
    expect(links.length).toBe(2);
    expect(links[0].prop("href")).toBe("/survey/create");
    expect(links[0].children().text()).toBe("Create Survey");
    expect(links[1].prop("href")).toBe("/dashboard");
    expect(links[1].children().text()).toBe("My Surveys");
  });

  it("contains logo", () => {
    const logo = wrapper.find(".logo");
    expect(logo.exists()).toBeTruthy();
    expect(logo.childAt(0).text()).toBe("X Choice");
  });
});
