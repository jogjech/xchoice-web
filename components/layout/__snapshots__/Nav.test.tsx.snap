// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar renders correctly 1`] = `
<Fragment>
  <Header>
    <Link
      href="/"
    >
      <div
        className="logo"
      >
        <Image
          alt="X Choice"
          className="logoImg"
          height="50"
          src="/logo.png"
          width="50"
        />
        <span
          className="logoText"
        >
          Choice
        </span>
      </div>
    </Link>
    <Menu
      mode="horizontal"
      theme="dark"
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "float": "right",
          }
        }
      >
        <ForwardRef(MenuOutlined)
          style={
            Object {
              "color": "white",
            }
          }
        />
      </div>
    </Menu>
  </Header>
  <DrawerWrapper
    closable={false}
    onClose={[Function]}
    placement="right"
    title=""
    visible={false}
  >
    <div
      className="drawerItem"
    >
      <Link
        href="/survey/create"
      >
        Create Survey
      </Link>
    </div>
    <div
      className="drawerItem"
    >
      <Link
        href="/dashboard"
      >
        My Surveys
      </Link>
    </div>
    <Divider
      dashed={true}
    />
    <div
      className="drawerItem"
    >
      <a
        href=""
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </DrawerWrapper>
</Fragment>
`;
